### PR TITLE
📚 Update documentation best practices

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Updates the "updating documentation" section of the docs to add best practices and guidlines for documentation.

### CODE REVIEW

File: docs/contributing-to-airbyte/updating-documentation.md [hunk 1]
```diff
@@ -20,3 +20,90 @@ If you're adding new files, don't forget to update `docs/SUMMARY.md`.
...
+### READMEs
+Every module should have a README containing:
+
+* A brief description of the module
+* development pre-requisites (like which language or binaries are required for development)
+* how to install dependencies
+* how to build and run the code locally & via Docker
+* any other information needed for local iteration
+  
+### Changelogs
```
Comment (avaidyanatha) [inline docs/contributing-to-airbyte/updating-documentation.md:43]: We should have people use this format for consistency:

## Changelog

```
| Version | Date       | Pull Request | Subject |
| :------ | :--------  | :-----       | :------ |
| 0.2.0   | 20XX-05-XX | [PR2#](https://github.com/airbytehq/airbyte/pull/PR2#) | description |
| 0.1.0   | 20XX-04-XX | [PR#](https://github.com/airbytehq/airbyte/pull/PR#) | description |
```

File: docs/contributing-to-airbyte/updating-documentation.md [hunk 2]
```diff
@@ -20,3 +20,90 @@ If you're adding new files, don't forget to update `docs/SUMMARY.md`.
...
+* READMEs
+* Changelogs
+* Github Issues & Pull Requests
+* Source code comments
+* How-to guides
+
+Below are some best practices related to each of these. 
+
+### READMEs
+Every module should have a README containing:
```
Comment (avaidyanatha) [inline docs/contributing-to-airbyte/updating-documentation.md:35]: Maybe a section on setting up the source/destination in the UI, explaining the fields?

File: docs/contributing-to-airbyte/updating-documentation.md [hunk 3]
```diff
@@ -20,3 +20,90 @@ If you're adding new files, don't forget to update `docs/SUMMARY.md`.
...
+
+_subpar PR title_: `Update tests`. Which tests? What was the update?
+  
+_good PR title_: `Source MySQL: update acceptance tests to connect to SSL-enabled database`. Specific about the domain and change that was made. 
+
+**PR title conventions**
+When creating a PR, follow the naming conventions depending on the change being made: 
+
+* New connectors: “🎉 New source or destination: <name>” e.g: `🎉 New Source: Okta`
+* New features: “🎉<Source or Destination> <name>: <feature description> E.g:
```
Comment (avaidyanatha) [inline docs/contributing-to-airbyte/updating-documentation.md:86]: Add a small note about core features?

File: docs/contributing-to-airbyte/updating-documentation.md [hunk 4]
```diff
@@ -20,3 +20,90 @@ If you're adding new files, don't forget to update `docs/SUMMARY.md`.
...
+*  Any refactors, cleanups, etc.. that are not visible improvements to the user should not have emojis
+
+The emojis help us identify which commits should be included in the product release notes. 
+
+#### Descriptions 
+**Context**: Provide enough information (or a link to enough information) in the description so team members with no context can understand what the issue or PR is trying to accomplish. This usually means you should include two things: 
+
+1. Some background information motivating the problem
+2. A description of the problem itself
+
```
Comment (avaidyanatha) [inline docs/contributing-to-airbyte/updating-documentation.md:101]:
```suggestion
2. A description of the problem itself
3. Good places to start reading and file changes that can be skipped
```

**Review by avaidyanatha — COMMENTED** [review]
Looks good, some comments.

File: docs/contributing-to-airbyte/updating-documentation.md (hunk 1 above)
Comment (marcosmarxm) [reply docs/contributing-to-airbyte/updating-documentation.md:43]: @sherifnada Abhi and I are adding the changelog into the connector documentation.

File: docs/contributing-to-airbyte/updating-documentation.md (hunk 3 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/updating-documentation.md:86]: done

File: docs/contributing-to-airbyte/updating-documentation.md (hunk 1 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/updating-documentation.md:43]: done

File: docs/contributing-to-airbyte/updating-documentation.md (hunk 2 above)
Comment (avaidyanatha) [reply docs/contributing-to-airbyte/updating-documentation.md:35]: Agree. If this is just GH facing then I'm fine with that.

**Pull request merged**